flocker/acceptance/testtools.py

```diff
@@ -544,21 +600,34 @@ def delete_items(items):
             )
             return get_items
 
-        def cleanup_containers():
+        def cleanup_containers(_):
             return api_clean_state(
                 self.configured_containers,
                 self.current_containers,
                 lambda item: self.remove_container(item[u"name"]),
             )
 
-        def cleanup_datasets():
+        def cleanup_datasets(_):
             return api_clean_state(
                 self.client.list_datasets_configuration,
                 self.client.list_datasets_state,
                 lambda item: self.client.delete_dataset(item.dataset_id),
             )
 
-        return cleanup_containers().addCallback(lambda _: cleanup_datasets())
+        def cleanup_leases():
+            get_items = self.client.list_leases()
+
+            def release_all(leases):
+                for lease in leases:
+                    self.client.release_lease(lease.dataset_id)
```

Comment:
Shouldn't this gather the results of release_lease into a deferred and return that? i.e. is it a problem that this doesn't wait for the leases to be released?

If it isn't a problem, then please add a comment explaining why.